codersdk/deployment.go

```diff
@@ -830,6 +843,20 @@ when required by your organization's security policy.`,
 			Value:       &c.DERP.Config.Path,
 			Group:       &deploymentGroupNetworkingDERP,
 			YAML:        "configPath",
+			// --derp-config-url supports file: URLs now, so this is hidden.
+			Hidden: true,
+		},
+		{
+			Name:        "Use Tailscale DERP network",
+			Description: "Use Tailscale's global relay and STUN network for your DERP mapping. The embedded relay will also be included if it is enabled. The STUN servers set by `--derp-server-stun-addresses` will be added to the default region instead of their own regions. This value will be ignored if you set a `--derp-config-url`.",
```

Comment:
This comes across as too complicated to me.  I really think we should divide the world into 2 camps:

1. People that just want stuff working and don't care that much --- they get tailscale DERP map, plain and simple.
2. People that have specific requirements about what is allowed to talk to what --- they get all these other flags like custom DERP regions and STUN addresses (or disabling STUN/p2p).

We shouldn't have some option where we take the Tailscale DERP map and then try to augment it with custom stuff.  If operators want that level of customization, they should just grab the tailscale maps and then customize it directly, and pass it in with `CODER_DERP_CONFIG_URL`.

We could (eventually) make it default enabled if you _don't_ have any of the other flags that customize DERP/STUN set.

Reply:
Is your concern about us adding the Coder default region to the map as well? I left it like that because I don't want to change the "built-in DERP server" flag to be default disabled.

Reply:
I'm concerned with both the embedded DERP and adding additional STUN servers in the default region.

Why do we want to keep the built-in DERP server by default when people use Tailscale?